src/TriggerBinding/SqlTriggerTargetScaler.cs

```diff
@@ -14,24 +17,80 @@ internal sealed class SqlTriggerTargetScaler : ITargetScaler
         private readonly int _maxChangesPerWorker;
+        private readonly ILogger _logger;
+        private readonly string _connectionString;
+        private readonly SqlObject _userTable;
+        private static DateTime _lastTableCheck = DateTime.MinValue;
+
 
         public SqlTriggerTargetScaler(string userFunctionId, SqlObject userTable, string userDefinedLeasesTableName, string connectionString, int maxChangesPerWorker, ILogger logger)
         {
             this._metricsProvider = new SqlTriggerMetricsProvider(connectionString, logger, userTable, userFunctionId, userDefinedLeasesTableName);
             this.TargetScalerDescriptor = new TargetScalerDescriptor(userFunctionId);
             this._maxChangesPerWorker = maxChangesPerWorker;
+            this._logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this._connectionString = !string.IsNullOrEmpty(connectionString) ? connectionString : throw new ArgumentNullException(nameof(connectionString));
+            this._userTable = userTable ?? throw new ArgumentNullException(nameof(userTable));
         }
 
         public TargetScalerDescriptor TargetScalerDescriptor { get; }
 
         public async Task<TargetScalerResult> GetScaleResultAsync(TargetScalerContext context)
         {
-            SqlTriggerMetrics metrics = await this._metricsProvider.GetMetricsAsync();
+            try
+            {
+                SqlTriggerMetrics metrics = await this._metricsProvider.GetMetricsAsync();
+
+                // Instance concurrency value is set by the functions host when dynamic concurrency is enabled. See https://learn.microsoft.com/en-us/azure/azure-functions/functions-concurrency for more details.
+                int concurrency = context.InstanceConcurrency ?? this._maxChangesPerWorker;
 
-            // Instance concurrency value is set by the functions host when dynamic concurrency is enabled. See https://learn.microsoft.com/en-us/azure/azure-functions/functions-concurrency for more details.
-            int concurrency = context.InstanceConcurrency ?? this._maxChangesPerWorker;
+                return GetScaleResultInternal(concurrency, metrics.UnprocessedChangeCount);
+            }
+            catch (Exception ex)
+            {
+                // If the exception is SQL exception and indicates that the object name is invalid, it means that the global state and leases table are not created
+                // Check for the error number 208 or the error message https://learn.microsoft.com/en-us/sql/relational-databases/errors-events/mssqlserver-208-database-engine-error?view=sql-server-ver17
+                if (ex is SqlException sqlEx && (sqlEx.Number == 208 || sqlEx.Message.Contains("Invalid object name") ||
+                                   (sqlEx.InnerException != null &&
+                                    sqlEx.InnerException.Message.Contains("Invalid object name"))))
```

Comment:
Is there a reason to keep the message checking? That seems risky - we have no idea what the errors from the engine could contain - it's possible (albeit unlikely) that some other random error or object name could happen to contain the string "invalid object name" and cause problems.